File: src/sweetdreams.lua
xb, yb = io.read("*n", "*n")
n = io.read("*n")
safe = true
for _ = 1, n do
    xa, ya = io.read("*n", "*n")
    if (xa - xb) * (xa - xb) + (ya - yb) * (ya - yb) <= 64 then
        safe = false
        break
    end
end
if safe then
    print('YES')
else
    print('NO')
end

File: src/sweetdreams.py
xb, yb = [int(d) for d in input().split()]
n = int(input())
safe = True
for _ in range(n):
    xa, ya = [int(d) for d in input().split()]
    if (xa - xb) ** 2 + (ya - yb) ** 2 <= 64:
        safe = False
        break
print("YES" if safe else "NO")
